first_set_of_refactorings/extract_variable/example_in_a_class.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+# formerly: Introduce Explaining Variables
+# inverse of: Inline Variable
+# @see https://memberservices.informit.com/my_account/webedition/9780135425664/html/extractfunction.html
+
+# Goal: Add a name to an expression
+
+module FirstSetOfRefactorings
+  module ExtractVaiable
+    module ExampleInAClass
+      class BeforeRefactor < RefactorBase
+        attr_accessor :quantity
+        attr_accessor :item_price
+
+        def initialize(quantity, item_price)
+          super()
+
+          @quantity = quantity
+          @item_price = item_price
+        end
+
+        def price
+          # price is base price - quantity discount + shipping
+          return base_price - [0, @quantity - 500].max * @item_price *
```

Comment:
Where is `base_price` coming from? If you create tests, you would get an error.

The `return` is superfluous